### PR TITLE
fix: resolve ui issues,add currency-aware-donation fallback and migrate history to DataTable

### LINKED ISSUE #7134: Several Issues - User Portal Donations Screen
**Describe the bug**

https://github.com/user-attachments/assets/0f35122e-3401-4b30-a291-f64217ecd0d5

**To Reproduce**

1. Login to the user portal
2. Select an organization
3. Select the donate tab

**Expected behavior**

- The issues seen in the screen shots and videos need to be rectified

**Screenshots**

Image

### PR DESCRIPTION
**Pull request by cotishq**
**What kind of change does this PR introduce?**

This PR introduces a **bug fix + UI refactor + compatibility improvement**:
- **Bug fixes**: layout overlap, unnecessary UI content, button alignment, and incorrect API-unavailable toast behavior.
- **Refactor**: migrated donation history rendering to the shared **DataTable** component.
- **Compatibility improvement**: added currency-aware donation mutation flow with safe fallback to legacy mutation for older API schema versions.

**Issue Number:**

Fixes #7134 

**Snapshots/Videos:**
Before:
Screenshot From 2026-02-11 17-53-49

After:
image

video:

https://github.com/user-attachments/assets/4f9c7ee6-eac3-4cac-9b50-daaadfe82fd7

**If relevant, did you update the documentation?**

No

**Summary**

This PR resolves the reported Donate page UI/UX issues in `talawa-admin` and aligns the donation history section with the shared table standard.

## Motivation

The Donate screen had multiple regressions and inconsistencies:
- Search bar/layout overlap issues.
- Unwanted right-side Members/Events panel on the page.
- Unnecessary helper text below amount input.
- Donate button alignment mismatch.
- False “API server unavailable” toast shown even when API was running.
- Legacy donation history rendering (cards + separate pagination) instead of the shared `DataTable`.
- Currency selection did not reliably propagate into donation mutation flow when backend supported currency input.

These fixes improve usability, consistency with repo patterns, and make the screen production-ready for the assigned issue.

### UI fixes (Donate screen)
- Fixed search bar overlap and container layout behavior.
- Removed the right-side Members/Events section from the Donate page.
- Removed the unnecessary text:  
  `"Please enter the numerical value for the donation amount"`.
- Moved/aligned the Donate button to the right as requested.

### API error toast behavior
- Updated global error handling (`src/index.tsx` + tests) to avoid false “API unavailable” notifications for non-connection GraphQL failures.

### Currency-aware donation mutation
- Added and used currency-capable mutation:
  - `DONATE_TO_ORGANIZATION_WITH_CURRENCY`
- Donation submission now:
  1. Tries currency-aware mutation first.
  2. Falls back to legacy `DONATE_TO_ORGANIZATION` if backend schema doesn’t support `currencyCode`.
- Refetch behavior updated to stay stable on currency change.

### Donation history migration to new DataTable
- Replaced legacy donation cards list rendering with shared `DataTable` in Donate page.
- Kept pagination functional.
- Kept search functional for donation history.
- Amount display now reflects selected currency symbol in the table UI.

### Design token compliance
- Removed hardcoded color values in `Donate.module.css`.
- Replaced with existing design tokens to satisfy pre-commit token validation.

## Testing / Validation

- Updated `Donate.spec.tsx` for currency-aware mutation + fallback behavior.
- Verified Donate tests pass locally.
- Verified token validator passes for changed files.

## Talawa-API Update Required

To fully complete currency behavior end-to-end, `talawa-api` must support donation currency in schema/resolvers (e.g., `currencyCode` input/field for `createDonation` and donation query output where applicable).  
Current admin fallback logic prevents breakage, but full “currency query/modify” correctness depends on backend support.

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Other information**

- Scope is limited to `talawa-admin` Donate flow and related global error handling/tests.
- Changes are backward-compatible on the frontend because donation submit now supports graceful fallback when API currency schema is unavailable.
- No breaking route or navigation changes were introduced.
- Styling updates are design-token compliant and pass token validation checks.
- Follow-up backend work in `talawa-api` is recommended to enable full currency-aware query/mutation behavior without fallback.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **New Features**
  * Currency-aware donation flow with selectable currencies, fallback to legacy flow, and donation history shown as a searchable, paginated table (includes donor and date).

* **Bug Fixes**
  * Smarter network-error gating to suppress spurious toasts for subscriptions, aborted requests, and distinguish 4xx vs 5xx responses.

* **Tests**
  * Expanded tests for currency flows and nuanced network error scenarios.

* **Style**
  * Donate UI layout/CSS refinements and new transition timing tokens.

* **Documentation**
  * Added donation translation keys (en/es/fr/hi/zh).

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please addrees these extensive findings:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/7203#pullrequestreview-3786717279

**Review by abhishekblue — COMMENTED** [review]
Based on the screenshot in the PR description, here are a few things I noticed that could improve the visual appeal a bit:

1. There's a noticeable gap between the "Donations" heading and the search bar, and again between the search bar and the donate card. Reducing the vertical spacing would make the layout feel tighter.
2. The donate button sits below the amount field and off to the right. It could either go inline with the amount field, or if kept below, right-aligned with proper spacing. Either way would look cleaner.
3. The "rows per page" and "0-0 of 0" pagination controls are visible even when there are no donations to show. It might be better to hide them when the table is empty.
4. One more small thing that could be good to verify would be to check if the search bar width matches how it appears on other screens? It would be nice to keep it consistent with the rest of the app.

File: src/style/tokens/transitions.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+:root {
+  --transition-fast: 0.2s;
+  --transition-normal: 0.3s;
```
Comment (abhishekblue) [inline src/style/tokens/transitions.css]: `--transition-normal` is defined but isn't used anywhere in this PR. Might be worth removing it to keep things clean. We can always add it later when it's actually needed.

**Comment by cotishq** [comment]
> Based on the screenshot in the PR description, here are a few things I noticed that could improve the visual appeal a bit:
> 
> 1. There's a noticeable gap between the "Donations" heading and the search bar, and again between the search bar and the donate card. Reducing the vertical spacing would make the layout feel tighter.
> 2. The donate button sits below the amount field and off to the right. It could either go inline with the amount field, or if kept below, right-aligned with proper spacing. Either way would look cleaner.
> 3. The "rows per page" and "0-0 of 0" pagination controls are visible even when there are no donations to show. It might be better to hide them when the table is empty.
> 4. One more small thing that could be good to verify would be to check if the search bar width matches how it appears on other screens? It would be nice to keep it consistent with the rest of the app.

ok i will address these all , thanks for the suggestions

File: src/index.tsx [hunk 2]
```diff
@@ -158,6 +158,34 @@ const errorLink = onError(
...
+        errorMessage.includes('network error') ||
+        (typeof statusCode === 'number' && statusCode >= 500);
+
+      if (
+        isSubscriptionOperation ||
+        isAbortOrCancelError ||
+        !isServerUnavailableError
+      ) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (cotishq) [reply src/index.tsx:187]: i have fixed this in the latest commit

File: src/screens/UserPortal/Donate/Donate.module.css [hunk 3]
```diff
@@ -77,10 +89,10 @@
...
   box-shadow:
     0 var(--shadow-offset-xs) var(--shadow-blur-sm) var(--shadow-spread-none)
-      rgba(168, 199, 250, 1),
+      var(--color-blue-200),
     0 var(--shadow-offset-md) var(--shadow-blur-lg) var(--shadow-spread-sm)
-      rgba(60, 64, 67, 0.15);
-  transition: box-shadow 0.2s ease;
+      var(--color-gray-700);
+  transition: box-shadow var(--transition-fast) ease;
 }
```
[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.module.css:95]: i have fixed this in the latest commit

File: src/screens/UserPortal/Donate/Donate.spec.tsx [hunk 4]
```diff
@@ -512,6 +529,54 @@ describe('Donate Component', () => {
...
+
+    await waitFor(() => {
+      expect(consoleErrorSpy).toHaveBeenCalledWith(
+        'Missing required donation identifiers for mutation.',
+        expect.objectContaining({ userId: '123', organizationId: undefined }),
+      );
+    });
+    expect(mockToast.success).not.toHaveBeenCalled();
+    expect(mockErrorHandler).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.spec.tsx:584]: i have fixed this in the latest commit

File: src/screens/UserPortal/Donate/Donate.tsx [hunk 5]
```diff
@@ -83,7 +158,72 @@ export default function Donate(): JSX.Element {
...
+        'Currency-aware donation unsupported by backend, falling back to legacy mutation',
+        {
+          error,
+          combinedMessage,
+        },
+      );
+    }
+
+    return shouldFallback;
+  };
```
[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.tsx:208]: i have fixed this in the latest commit

File: src/screens/UserPortal/Donate/Donate.tsx [hunk 6]
```diff
@@ -7,31 +7,39 @@ import { useQuery, useMutation } from '@apollo/client';
...
   value: currency,
   label: currency,
 }));
 
+interface IDonationTableRow {
+  id: string;
+  donor: string;
+  amount: number;
+  updatedAt: string;
+}
```
[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.tsx]: this is fixed in the latest commit

File: src/screens/UserPortal/Donate/Donate.spec.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.spec.tsx:584]: Implemented. In both missing-identifier tests, I moved the negative assertions into the same [await waitFor(...)](https://file+.vscode-resource.vscode-cdn.net/home/cotishq/.cursor/extensions/openai.chatgpt-0.4.74-universal/webview/#) block as the console.error expectation to avoid async race conditions.

Updated:

[Donate.spec.tsx (lines 548-555)](https://file+.vscode-resource.vscode-cdn.net/home/cotishq/.cursor/extensions/openai.chatgpt-0.4.74-universal/webview/#)
[Donate.spec.tsx (lines 570-577)](https://file+.vscode-resource.vscode-cdn.net/home/cotishq/.cursor/extensions/openai.chatgpt-0.4.74-universal/webview/#)
Now each test waits for stabilization before asserting:

expect(mockToast.success).not.toHaveBeenCalled()
expect(mockErrorHandler).not.toHaveBeenCalled()

[the remark this answers is not in the document]
Comment (cotishq) [reply src/screens/UserPortal/Donate/Donate.spec.tsx:584]: image

i have pushed this change too, why you are addressing it again ? , please check the diff once again

**Comment by cotishq** [comment]
@palisadoes i have addressed all the cr changes and @abhishekblue changes too , all ci checks are passing except ignore-sensitive-files one , please review it and let me know what other changes are required to be done

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Please upload a video of your work
    - The margins don't look equal on the screenshots
1. Please address these extensive findings:
     - https://github.com/PalisadoesFoundation/talawa-admin/pull/7203#issuecomment-3905279565

**Comment by cotishq** [comment]
> 1. Please upload a video of your work
>    
>    * The margins don't look equal on the screenshots
> 2. Please address these extensive findings:
>    
>    * [#7203 (comment)](https://github.com/PalisadoesFoundation/talawa-admin/pull/7203#issuecomment-3905279565)

@palisadoes  i have addressed all the extensive findings and also uploaded a video of my work in pr description , please check it out and let me know what else changes are required

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Review by palisadoes — CHANGES_REQUESTED** [review]
A great improvement. Please address these issues:

---

Image

---

Image

---

Image

---

Image


---

Image

**Comment by cotishq** [comment]
> A great improvement. Please address these issues:
> 
> Image
> Image
> Image
> Image
> Image

@palisadoes  I have fixed all the referrenced ui alignment and empty state issues . 
Also Updated PR description with latest screenshot and video, 
Regarding your concern about the pop up stating , the submitted information is invalid is not a frontend bug , it is related to backend 
1) createDonation fails with HTTP 400 in user portal even for valid amount input.
2) Frontend currently sends placeholder payPalId: 'paypalId' (not real transaction id).
3) Error handler collapses all 400s into generic toast, masking actual GraphQL validation message.

suggested fix path: 
1) Backend: clarify/relax payPalId requirement (or provide test-mode donation endpoint).
2) Frontend: send real payment transaction ID (or required field per API contract).
3) Frontend: surface backend graphQLErrors[0].message instead of only generic error400 so debugging is clearer.

**Comment by palisadoes** [comment]
> > A great improvement. Please address these issues:
> > Image
> > Image
> > Image
> > Image
> > Image
> 
> @palisadoes I have fixed all the referrenced ui alignment and empty state issues . Also Updated PR description with latest screenshot and video, Regarding your concern about the pop up stating , the submitted information is invalid is not a frontend bug , it is related to backend
> 
>     1. createDonation fails with HTTP 400 in user portal even for valid amount input.
> 
>     2. Frontend currently sends placeholder payPalId: 'paypalId' (not real transaction id).
> 
>     3. Error handler collapses all 400s into generic toast, masking actual GraphQL validation message.
> 
> 
> suggested fix path:
> 
>     1. Backend: clarify/relax payPalId requirement (or provide test-mode donation endpoint).
> 
>     2. Frontend: send real payment transaction ID (or required field per API contract).
> 
>     3. Frontend: surface backend graphQLErrors[0].message instead of only generic error400 so debugging is clearer.

Thanks.

1. Please create a series of issues to fix the failure. 
2. How can this be done so that it works with any backend payment gateway?

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Revert the CSS token files.
2. Don't have transitions. This needs to be part of a broader CSS strategy.

**Comment by cotishq** [comment]
> 1. Revert the CSS token files.
> 2. Don't have transitions. This needs to be part of a broader CSS strategy.

@palisadoes  I have reverted the css token file, as per your concern

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Review by palisadoes — APPROVED** [review]
